src/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,64 @@
+import {getDocument, getSetImmediate, newMutationObserver} from './helpers'
+
+function waitForElementToBeRemoved(
+  callback,
+  {
+    container = getDocument(),
+    timeout = 4500,
+    mutationObserverOptions = {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      characterData: true,
+    },
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    if (typeof callback !== 'function') {
+      reject(
+        'waitForElementToBeRemoved requires a function as the first parameter',
+      )
+    }
+
+    // Check if the element is not present synchronously,
+    // As the name waitForElementToBeRemoved should check `present` --> `removed`
+    try {
+      const result = callback()
+      if (!result) {
```

Comment:
Should we be more specific in checking that the result is a single element (or non-empty array of elements)? If it's not then the error can be more specific as well: "The callback function which was passed did not return an element or non-empty array of elements. waitForElementToBeRemoved requires that the element(s) exist before waiting for removal."